waspc/cli/src/Wasp/Cli/Command/Deps.hs

```diff
@@ -18,17 +19,18 @@ deps =
     putStrLn $
       unlines $
         [ "",
-          title "Below are listed the dependencies that Wasp uses in your project. You can import and use these directly in the code as if you specified them yourself, but you can't change their versions.",
+          title "Below are the dependencies that this version of Wasp uses in projects. You can import and use these directly in the code as if you specified them yourself, but you can't change their versions.",
           ""
         ]
           ++ printDeps
             "Server dependencies:"
-            ( N.waspDependencies ServerGenerator.npmDepsForWasp
+            ( N.waspDependencies $ ServerGenerator.npmDepsForWasp Nothing
             )
+          ++ [printDep pgBossDependency ++ " <-- Only if PgBoss is used in your jobs"]
```

Comment:
How should we handle displaying conditional includes that depend AppSpec? Example: `pg-boss`, but only when `jobs` are used.

Reply:
Can we do it based on AppSpec?

Reply:
At this level of a call (in CLI), we won't have access to any AppSpec info I believe.

Reply:
True, but we could have if we needed to, right?
Check Command/Info.hs -> it actually parses the Wasp code and then uses some of the information from it. You could do the same thing, or go step forward and instead obtain actually AppSpec. So when somebody invokes list of dependencies, you would do the compile first, but only analyze phase, not generate, obtain AppSpec that way, and then use that.
I am not 100% sure if that is as easy as it sounds though, sounds doable to me now but it might be trickier hm. But maybe worth investigating as a direction?

Reply:
I also think such solution is most useful, because it really shows which dependencies are used -> otherwise they have do figure out themselves which of the optional dependencies are used.

Reply:
Good thinking, I'd love to have AppSpec here! Let me give it a go thnx 👍🏻

Reply:
Was able to extract it from compile and get an AppSpec in deps! 👍🏻